mediorum/server/migrate_qm.go

```diff
@@ -0,0 +1,173 @@
+package server
+
+import (
+	"context"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/jackc/pgx/v5"
+)
+
+func (ss *MediorumServer) startQmCidMigration() {
+	// Wait 1m after boot to start repair
+	time.Sleep(time.Minute)
+
+	logger := ss.logger.With("task", "migrate_qm")
+
+	// Repeat the migration process every minute in case new uploads legacy came in (they shouldn't).
+	// Subsequent runs should be very fast since the migration deletes rows from Files after adding them to the bucket and inserting a row into the blobs table.
+	for {
+		start := time.Now()
+
+		// TODO: It would be nice if this could be an optimizely flag
+		if ss.Config.MigrateQmCids {
+			logger.Info("Qm CID migration starting")
+			err := ss.migrateQmCids()
+			took := time.Since(start)
+			if err != nil {
+				logger.Error("Qm CID migration failed", "err", err, "took", took)
+			} else {
+				logger.Info("Qm CID migration OK", "took", took)
+			}
+		} else {
+			logger.Info("Qm CID migration disabled")
+		}
+
+		logger.Info("Qm CID migration sleeping", "sleep", time.Minute)
+		time.Sleep(time.Minute)
+	}
+}
+
+func (ss *MediorumServer) migrateQmCids() error {
+	pageSize := 100
+	var lastCreatedAt time.Time
+
+	for {
+		ctx := context.Background()
+		logger := ss.logger.With("task", "migrate_qm", "lastCreatedAt", lastCreatedAt)
+
+		/* assumes unique fileUUID for each row - check with:
+		SELECT *
+		FROM "Files"
+		WHERE "fileUUID" IN (
+				SELECT "fileUUID"
+				FROM "Files"
+				GROUP BY "fileUUID"
+				HAVING COUNT("fileUUID") > 1
+		)
+		*/
+
+		// note that fileUUID doesn't have an index, so we select by createdAt and delete by fileUUID
+		baseSql := `SELECT "storagePath", "multihash", "dirMultihash", "fileName", "fileUUID", "createdAt"
+    FROM "Files"
+    WHERE "type" IN ('copy320', 'image')`
```

Comment:
need to check null here as well for super legacy images. example row from prod usermetadata:

```
audius_creator_node=> select * from "Files" where multihash = 'QmNjJv1wQf2DJq3GNXjXzSL8UXFUGXfchg4NhL7UpbnF1f';
-[ RECORD 1 ]-----+-----------------------------------------------------------------------
multihash         | QmNjJv1wQf2DJq3GNXjXzSL8UXFUGXfchg4NhL7UpbnF1f
sourceFile        | 
storagePath       | /file_storage/files/nF1/QmNjJv1wQf2DJq3GNXjXzSL8UXFUGXfchg4NhL7UpbnF1f
createdAt         | 2019-10-08 22:52:08.597+00
updatedAt         | 2022-12-08 22:44:28.332+00
fileUUID          | 4b3f0b27-914a-4e81-b872-d0da8ee55070
cnodeUserUUID     | 5ff4b194-dd6b-4282-a454-352f9c737ea9
type              | null
fileName          | 
dirMultihash      | 
trackBlockchainId | 
clock             | 1
skipped           | f
```